File: temp/ComputerStore/library/src/Goods/Laptop.cpp
#include "Goods/Laptop.h"

Laptop::Laptop(double price, int ram, int ssdCapacity, processor cpu, double screenSize, int usbAmount, bool camera)
        : Computer(price, ram, ssdCapacity, cpu), screenSize(screenSize), camera(camera), USBAmount(usbAmount) {}

Laptop::~Laptop() {}

double Laptop::getScreenSize() const {
    return screenSize;
}

bool Laptop::isCamera() const {
    return camera;
}

int Laptop::getUSBAmount() const {
    return USBAmount;
}

double Laptop::getPrice() {
    return Computer::getPrice() + screenSize*10 + USBAmount*100 + (camera ? 200 : 0);
}

std::string Laptop::toString() {
    return Computer::toString() +
           "\nScreen Size: " + std::to_string(getScreenSize()) +
           "\nCamera: " + std::to_string(isCamera()) +
           "\nUSB Amount: " + std::to_string(getUSBAmount()) +
           "\nPrice: " + std::to_string(getPrice());
}

File: src/main/java/model/goods/Laptop.java
package model.goods;

import controller.exceptions.GoodException;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.ToString;

import java.util.Objects;
import java.util.UUID;

@Data
@EqualsAndHashCode(callSuper = true)
@ToString(callSuper = true)
public class Laptop extends Computer {
    private double screenSize;
    private boolean hasCamera;
    private int usbAmount;

    public Laptop() {
        super();
    }

    public Laptop(double basePrice, String goodName, int ram, int ssdCapacity, double screenSize, boolean hasCamera, int usbAmount, int count) throws GoodException {
        super(basePrice, goodName, ram, ssdCapacity, count);
        if (screenSize < 0) {
            throw new GoodException("Screen size cannot be negative.");
        }
        if (usbAmount < 0) {
            throw new GoodException("USB amount cannot be negative.");
        }
        this.screenSize = screenSize;
        this.hasCamera = hasCamera;
        this.usbAmount = usbAmount;
    }

    public double getPrice() {
        return getBasePrice() + screenSize * 10 + usbAmount * 100 + (hasCamera ? 200 : 0);
    }


}
